src/storage/iceberg_insert.cpp

```diff
@@ -0,0 +1,386 @@
+// Plan
+//===--------------------------------------------------------------------===//
+static optional_ptr<CopyFunctionCatalogEntry> TryGetCopyFunction(DatabaseInstance &db, const string &name) {
+	D_ASSERT(!name.empty());
+	auto &system_catalog = Catalog::GetSystemCatalog(db);
+	auto data = CatalogTransaction::GetSystemTransaction(db);
+	auto &schema = system_catalog.GetSchema(data, DEFAULT_SCHEMA);
+	return schema.GetEntry(data, CatalogType::COPY_FUNCTION_ENTRY, name)->Cast<CopyFunctionCatalogEntry>();
+}
+
+static Value GetFieldIdValue(const IcebergColumnDefinition &column) {
+	auto column_value = Value::BIGINT(column.id);
+	if (column.children.empty()) {
+		// primitive type - return the field-id directly
+		return column_value;
+	}
+	// nested type - generate a struct and recurse into children
+	child_list_t<Value> values;
+	values.emplace_back("__duckdb_field_id", std::move(column_value));
+	for (auto &child : column.children) {
+		values.emplace_back(child->name, GetFieldIdValue(*child));
+	}
+	return Value::STRUCT(std::move(values));
+}
+
+static Value WrittenFieldIds(const IcebergTableSchema &schema) {
+	auto &columns = schema.columns;
+
+	child_list_t<Value> values;
+	for (idx_t c_idx = 0; c_idx < columns.size(); c_idx++) {
+		auto &column = columns[c_idx];
+		values.emplace_back(column->name, GetFieldIdValue(*column));
+	}
+	return Value::STRUCT(std::move(values));
+}
+
+PhysicalOperator &IRCatalog::PlanInsert(ClientContext &context, PhysicalPlanGenerator &planner, LogicalInsert &op,
+                                        optional_ptr<PhysicalOperator> plan) {
+	if (op.return_chunk) {
+		throw BinderException("RETURNING clause not yet supported for insertion into Iceberg table");
+	}
+	if (op.action_type != OnConflictAction::THROW) {
+		throw BinderException("ON CONFLICT clause not yet supported for insertion into Iceberg table");
+	}
+
+	auto &table_entry = op.table.Cast<ICTableEntry>();
+	auto &table_info = table_entry.table_info;
+	auto iceberg_path = table_info.BaseFilePath();
+	auto &schema = table_info.table_metadata.GetLatestSchema();
+
+	auto &partition_spec = table_info.table_metadata.GetLatestPartitionSpec();
+	if (!partition_spec.IsUnpartitioned()) {
+		throw NotImplementedException("INSERT into a partitioned table is not supported yet");
+	}
+
+	// Create Copy Info
+	auto info = make_uniq<CopyInfo>();
+
+	auto data_path = iceberg_path + "/data";
+	info->file_path = data_path;
+	info->format = "parquet";
+	info->is_from = false;
+
+	vector<Value> field_input;
+	field_input.push_back(WrittenFieldIds(schema));
+	info->options["field_ids"] = std::move(field_input);
+
+	// Get Parquet Copy function
+	auto copy_fun = TryGetCopyFunction(*context.db, "parquet");
+	if (!copy_fun) {
+		throw MissingExtensionException("Did not find parquet copy function required to write to iceberg table");
+	}
+
+	vector<idx_t> partition_columns;
+	//! TODO: support partitions
```

Comment:
nit: there's quite a lot of commented out code here

Reply:
There is, because it's relevant for supporting partitions, which this PR doesn't do yet, but I think it's logically one of the next steps to support